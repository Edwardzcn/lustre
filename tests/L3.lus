
type T = struct { x : int; y : bool };

function f (x : T) returns (y : int)
let
  --%MAIN;
  y = if x.y then x.x else 0;
tel
