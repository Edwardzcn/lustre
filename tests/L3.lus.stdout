node (x.x_6 : int, x.y_7 : bool)
  returns (y_3)
  assumes ()
  shows ()
let
  y_3 = ITE(x.y_7, x.x_6, 0)
tel
--- Step 1 ---
x.x_6 : int  = 1
x.y_7 : bool  = True
y_3 = 1
--- Step 2 ---
x.x_6 : int  = 2
x.y_7 : bool  = False
y_3 = 0
--- Step 3 ---
x.x_6 : int  = 3
x.y_7 : bool  = True
y_3 = 3
--- Step 4 ---
x.x_6 : int  = 4
x.y_7 : bool  = False
y_3 = 0
--- Step 5 ---
x.x_6 : int  = 5
x.y_7 : bool  = False
y_3 = 0
--- Step 6 ---
x.x_6 : int  = 6
x.y_7 : bool  = True
y_3 = 6
